pyvisa_py/protocols/hislip.py

```diff
@@ -0,0 +1,714 @@
+    2: "successSharedLock",
+    3: "error",
+}
+LOCKRESPONSECONTROLCODE.update(
+    {value: key for (key, value) in LOCKRESPONSECONTROLCODE.items()}
+)
+
+REMOTELOCALCONTROLCODE: LookupTable = {
+    0: "disableRemote",
+    1: "enableRemote",
+    2: "disableAndGTL",
+    3: "enableAndGotoRemote",
+    4: "enableAndLockoutLocal",
+    5: "enableAndGTRLLO",
+    6: "justGTL",
+}
+REMOTELOCALCONTROLCODE.update(
+    {value: key for (key, value) in REMOTELOCALCONTROLCODE.items()}
+)
+
+HEADER_FORMAT = "!2sBBIQ"
+# !  = network order,
+# 2s = prologue ('HS'),
+# B  = message type (unsigned byte),
+# B  = control code (unsigned byte),
+# I  = message parameter (unsigned int),
+# Q  = payload length (unsigned long long)
+HEADER_SIZE = struct.calcsize(HEADER_FORMAT)
+
+
+class Struct(dict):
+    def __init__(self, **kwargs):
+        super(Struct, self).__init__(**kwargs)
+        self.__dict__ = self
+
+
+class Instrument:  # pylint: disable=too-many-instance-attributes
+    """
+    this is the principal export from this module.  it opens up a HiSLIP connection
+    to the instrument at the specified IP address.
+    """
+
+    def __init__(self, ip_addr, open_timeout=None, port=PORT):
+        # init transaction:
+        #     C->S: Initialize
+        #     S->C: InitializeResponse
+        #     C->S: AsyncInitialize
+        #     S->C: AsyncInitializeResponse
+
+        timeout = 1e-3 * (open_timeout or 1000)
+
+        # open the synchronous socket and send an initialize packet
+        self._sync = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self._sync.connect((ip_addr, port))
+        self._sync.settimeout(timeout)
+        self._sync.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
+        init = self.initialize()
+        if init.overlap != 0:
+            print("**** prefer overlap = %d" % init.overlap)
+
+        # open the asynchronous socket and send an initialize packet
+        self._async = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self._async.connect((ip_addr, port))
+        self._async.settimeout(timeout)
+        self._async.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
+        self._async_init = self.async_initialize(session_id=init.session_id)
+
+        # get the maximum message size
+        max_msg_size = self.async_maximum_message_size(1 << 24)
+        # print("HEADER_SIZE = %s, async_maximum_message_size = %s" % (HEADER_SIZE, max_msg_size))
+        # print("max_payload_size = %s" % (max_msg_size - HEADER_SIZE))
+        self.max_payload_size = max_msg_size - HEADER_SIZE
+
+        self.timeout = 10
+        self.rmt = 0
+        self.receiver = None
+        self.message_id = 0xFFFFFF00
+
+    # ================ #
+    # MEMBER FUNCTIONS #
+    # ================ #
+
+    def close(self):
+        self._sync.close()
+        self._async.close()
+
+    @property
+    def timeout(self):
+        """returns the timeout value in seconds for both the sync and async sockets"""
```

Comment:
A property is basically an attribute so I prefer the docstring to reflect this.
```suggestion
        """Timeout value in seconds for both the sync and async sockets"""
```